receiver/snowflakereceiver/client.go

```diff
@@ -13,3 +13,426 @@
 // limitations under the License.
 
 package snowflakereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snowflakereceiver"
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	sf "github.com/snowflakedb/gosnowflake"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+// queries
+var (
+	billingMetricsQuery          = "select SERVICE_TYPE, NAME, sum(CREDITS_USED_COMPUTE), sum(CREDITS_USED_CLOUD_SERVICES), sum(CREDITS_USED) from METERING_HISTORY where start_time >= DATEADD(hour, -24, current_timestamp()) group by 1, 2;"
+	warehouseBillingMetricsQuery = "select WAREHOUSE_NAME, sum(CREDITS_USED_COMPUTE), sum(CREDITS_USED_CLOUD_SERVICES), sum(CREDITS_USED) from WAREHOUSE_METERING_HISTORY where start_time >= DATEADD(hour, -24, current_timestamp()) group by 1;"
+	loginMetricsQuery            = "select USER_NAME, ERROR_MESSAGE, REPORTED_CLIENT_TYPE, IS_SUCCESS, count(*) from LOGIN_HISTORY where event_timestamp >= DATEADD(hour, -24, current_timestamp()) group by 1, 2, 3, 4;"
+	highLevelQueryMetricsQuery   = "select WAREHOUSE_NAME, AVG(AVG_RUNNING), AVG(AVG_QUEUED_LOAD), AVG(AVG_QUEUED_PROVISIONING), AVG(AVG_BLOCKED) from WAREHOUSE_LOAD_HISTORY where start_time >= DATEADD(hour, -24, current_timestamp()) group by 1;"
+	dbMetricsQuery               = "select SCHEMA_NAME, EXECUTION_STATUS, ERROR_MESSAGE, QUERY_TYPE, WAREHOUSE_NAME, DATABASE_NAME, WAREHOUSE_SIZE, USER_NAME, COUNT(QUERY_ID), AVG(queued_overload_time), AVG(queued_repair_time), AVG(queued_provisioning_time), AVG(TOTAL_ELAPSED_TIME), AVG(EXECUTION_TIME), AVG(COMPILATION_TIME), AVG(BYTES_SCANNED), AVG(BYTES_WRITTEN), AVG(BYTES_DELETED), AVG(BYTES_SPILLED_TO_LOCAL_STORAGE), AVG(BYTES_SPILLED_TO_REMOTE_STORAGE), AVG(PERCENTAGE_SCANNED_FROM_CACHE), AVG(PARTITIONS_SCANNED), AVG(ROWS_UNLOADED), AVG(ROWS_DELETED), AVG(ROWS_UPDATED), AVG(ROWS_INSERTED), AVG(COALESCE(ROWS_PRODUCED,0)) from QUERY_HISTORY where start_time >= DATEADD(hour, -24, current_timestamp()) group by 1, 2, 3, 4, 5, 6, 7, 8;"
+	sessionMetricsQuery          = "select USER_NAME, count(distinct(SESSION_ID)) from Sessions where created_on >= DATEADD(hour, -24, current_timestamp()) group by 1;"
+	snowpipeMetricsQuery         = "select pipe_name, sum(credits_used), sum(bytes_inserted), sum(files_inserted) from pipe_usage_history where start_time >= DATEADD(hour, -24, current_timestamp()) group by 1;"
+	storageMetricsQuery          = "select STORAGE_BYTES, STAGE_BYTES, FAILSAFE_BYTES from STORAGE_USAGE ORDER BY USAGE_DATE DESC LIMIT 1;"
+)
+
+// snowflake client is comprised of a sql.DB (the proper 'client' in question),
+// a connection string (dsn), a collection of queries (built from which metrics are enabled),
+// and a logger
+type defaultSnowflakeClient struct {
+	client *sql.DB
+	dsn    *string
+	logger *zap.Logger
+}
+
+var _ snowflakeClient = (*defaultSnowflakeClient)(nil)
+
+type snowflakeClient interface {
```

Comment:
Why do we need the interface? I don't see where it's implemented other than by `defaultSnowflakeClient`. I think it'll be simpler and easier to maintain if we keep just `snowflakeClient` struct unless we plan to have any other implementations